x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicy.java

```diff
@@ -322,7 +350,10 @@ public boolean equals(Object obj) {
             return false;
         }
         LifecyclePolicy other = (LifecyclePolicy) obj;
-        return Objects.equals(name, other.name) && Objects.equals(phases, other.phases) && Objects.equals(metadata, other.metadata);
+        return Objects.equals(name, other.name)
+            && Objects.equals(phases, other.phases)
+            && Objects.equals(metadata, other.metadata)
+            && isDeprecated() == other.isDeprecated();
```

Comment:
Same comment here about using the variable directly rather than using `isDeprecated()` to avoid thinking an update is noop when it's not.